### PR TITLE
[SPARK-22954][SQL] Fix the exception thrown by Analyze command on temporary views to throw AnalysisException

### PR DESCRIPTION
**Pull request by ghost**
## What changes were proposed in this pull request?
Previously running ANALYZE on temporary views produced NoSuchTableException even though the view was present. It is misleading. So it has to be changed to throw AnalysisException since it comes in the analysis part. 

## How was this patch tested?
Some unit tests were fixed to intercept AnalysisException instead of NoSuchTableException.

### CODE REVIEW

**Comment by cloud-fan** [comment]
ok to test

**Comment by cloud-fan** [comment]
what's the error message after this patch?

**Comment by cloud-fan** [comment]
can you fix the test?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala [hunk 1]
```diff
@@ -31,9 +31,9 @@ case class AnalyzeTableCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val sessionState = sparkSession.sessionState
-    val db = tableIdent.database.getOrElse(sessionState.catalog.getCurrentDatabase)
-    val tableIdentWithDB = TableIdentifier(tableIdent.table, Some(db))
-    val tableMeta = sessionState.catalog.getTableMetadata(tableIdentWithDB)
+    val db = tableIdent.database
+    val tableIdentWithDB = TableIdentifier(tableIdent.table, db)
+    val tableMeta = sessionState.catalog.getTempViewOrPermanentTableMetadata(tableIdentWithDB)
```
Comment (jiangxb1987) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala]: Wouldn't this fail if we have a table that neglect the current database in tableIdent?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala [hunk 2]
```diff
@@ -154,11 +155,17 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       assertNoSuchTable(s"TRUNCATE TABLE $viewName")
       assertNoSuchTable(s"SHOW CREATE TABLE $viewName")
       assertNoSuchTable(s"SHOW PARTITIONS $viewName")
-      assertNoSuchTable(s"ANALYZE TABLE $viewName COMPUTE STATISTICS")
-      assertNoSuchTable(s"ANALYZE TABLE $viewName COMPUTE STATISTICS FOR COLUMNS id")
+      assertAnalysisException(s"ANALYZE TABLE $viewName COMPUTE STATISTICS")
```
Comment (jiangxb1987) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala]: We should also check the error message to ensure the `AnalysisException` is not thrown from elsewhere.

**Comment by jiangxb1987** [comment]
We shall also update for `AnalyzePartitionCommand` and `AnalyzeColumnCommand`.

**Comment by cloud-fan** [comment]
actually it's not really useful to have `NoSuchTableException`, `NoSuchFunctionException`, etc. always using AnalysisException seems fine. CC @gatorsmile

**Comment by suchithjn225** [comment]
I did't understand why the build is getting killed.

**Comment by cloud-fan** [comment]
retest this please

**Comment by jiangxb1987** [comment]
Could you also add test cases that cover ANALYZE PARTITION and ANALYZE COLUMN queries?

**Comment by suchithjn225** [comment]
Done.

**Comment by gatorsmile** [comment]
ok to test

**Comment by gatorsmile** [comment]
cc @wzhfy

**Pull request closed without merging**